### PR TITLE
Fix `Intersect` with constant strings

### LINKED ISSUE #37727: `SELECT 'Play ClickHouse' InterSect SELECT 'Play ClickHouse'` makes null pointer dereference.

### PR DESCRIPTION
**Pull request by antonio2368**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix `SELECT ... INTERSECT` and `EXCEPT SELECT` statements with constant string types.

Fix #37727 

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Review by alesapin — APPROVED** [review]

File: src/Common/ColumnsHashing.h [hunk 1]
```diff
@@ -83,8 +85,11 @@ struct HashMethodString
 
     HashMethodString(const ColumnRawPtrs & key_columns, const Sizes & /*key_sizes*/, const HashMethodContextPtr &)
     {
-        const IColumn & column = *key_columns[0];
-        const ColumnString & column_string = assert_cast<const ColumnString &>(column);
+        const IColumn * column = key_columns[0];
+        if (isColumnConst(*column))
```
Comment (alesapin) [inline src/Common/ColumnsHashing.h:89]: BTW, what about other modifiers like `Nullable`?

**Review by alesapin — APPROVED** [review]

File: src/Common/ColumnsHashing.h (hunk 1 above)
Comment (antonio2368) [reply src/Common/ColumnsHashing.h:89]: That works, nullable columns use specific hashing method for integer and generic method for other types.
Single const string is a special case that uses String hash method but it didn't handle the const, and that was the case reported in the issue.

**Comment by tavplubix** [comment]
@alexey-milovidov it is **not** expected, you've just broken stress tests in master by merging this. To avoid failures like this `no-backward-compatibility-check` tag must be added (https://github.com/ClickHouse/ClickHouse/commit/6a516099152c8c14ffbcc6ffdc79c46077918889)

@Avogar, is it possible to make BC check more convenient? Maybe use git history to exclude tests that were created (or modified?) between HEAD and merge-base of HEAD and previous release tag...

**Comment by tavplubix** [comment]
Btw, there's a typo in the test name

**Comment by tavplubix** [comment]
> Also, isn't it better to add the tag with the version so we don't disable it indefinitely? I assume you just wanted to push a quick fix.


Yep, tag with the version is better. (however, we will  try to use git history instead)

**Pull request merged**